test/unit/messaging/batch-requests.spec.ts

```diff
@@ -0,0 +1,239 @@
+    expect(responses.length).to.equal(3);
+    responses.forEach((response) => {
+      expect(response.status).to.equal(200);
+      expect(response.data).to.deep.equal(responseObject);
+    });
+    checkOutgoingRequest(stub, requests);
+  });
+
+  it('should handle both success and failure HTTP responses in a batch', async () => {
+    const stub = sinon.stub(httpClient, 'send').resolves(
+      createMultipartResponse([responseObject, responseObject], [responseObject]));
+    stubs.push(stub);
+    const requests: SubRequest[] = [
+      {url: 'https://example.com', body: {foo: 1}},
+      {url: 'https://example.com', body: {foo: 2}},
+      {url: 'https://example.com', body: {foo: 3}},
+    ];
+    const batch = new BatchRequestClient(httpClient, batchUrl);
+
+    const responses: HttpResponse[] = await batch.send(requests);
+
+    expect(responses.length).to.equal(3);
+    responses.forEach((response, idx) => {
+      const expectedStatus = idx < 2 ? 200 : 500;
+      expect(response.status).to.equal(expectedStatus);
+      expect(response.data).to.deep.equal(responseObject);
+    });
+    checkOutgoingRequest(stub, requests);
+  });
+
+  it('should reject on top-level HTTP error responses', async () => {
+    const stub = sinon.stub(httpClient, 'send').rejects(
+      utils.errorFrom({error: 'test'}));
+    stubs.push(stub);
+    const requests: SubRequest[] = [
+      {url: 'https://example.com', body: {foo: 1}},
+      {url: 'https://example.com', body: {foo: 2}},
+      {url: 'https://example.com', body: {foo: 3}},
+    ];
+    const batch = new BatchRequestClient(httpClient, batchUrl);
+
+    try {
+      await batch.send(requests);
+      sinon.assert.fail('No error thrown for HTTP error');
+    } catch (err) {
+      expect(err).to.be.instanceOf(HttpError);
+      expect((err as HttpError).response.status).to.equal(500);
+      checkOutgoingRequest(stub, requests);
+    }
+  });
+
+  it('should add common headers to the batch requests', async () => {
+    const stub = sinon.stub(httpClient, 'send').resolves(
+      createMultipartResponse([responseObject]));
+    stubs.push(stub);
+    const requests: SubRequest[] = [
+      {url: 'https://example.com', body: {foo: 1}},
+    ];
+    const commonHeaders = {'X-Custom-Header': 'value'};
+    const batch = new BatchRequestClient(httpClient, batchUrl, commonHeaders);
+
+    const responses: HttpResponse[] = await batch.send(requests);
+
+    expect(responses.length).to.equal(1);
+    expect(stub).to.have.been.calledOnce;
+    const args: HttpRequestConfig = stub.getCall(0).args[0];
+    expect(args.headers).to.have.property('X-Custom-Header', 'value');
+  });
+
+  it('should add sub request headers to the payload', async () => {
+    const stub = sinon.stub(httpClient, 'send').resolves(
+      createMultipartResponse([responseObject]));
+    stubs.push(stub);
+    const requests: SubRequest[] = [
+      {url: 'https://example.com', body: {foo: 1}, headers: {'X-Custom-Header': 'value'}},
```

Comment:
The test uses the same header key/value for the common headers and the subrequests. it is better to use different values to confirm source of expected value.

Reply:
Done